### ISSUE TITLE
How to get center location and scale when inference?

### ISSUE THREAD

**Issue by jinfagang** [leoxiaobin/deep-high-resolution-net.pytorch#23]
I need to get the final result which seems can calling this functions:

```
def get_final_preds(config, batch_heatmaps, center, scale):

```

But it needs a center and a scale. How to get it anyway?

**Comment by Calmost** [comment]
Hi,have you ever got the final result ?

**Comment by Calmost** [comment]
I have get a center and a scale by using yolo detector. 
Could you show me your demo file to have inference from a given image.

**Comment by jinfagang** [comment]
@Calmost I don't think this 2 stage pose detector worth to do further inference, it does not make realtime

**Comment by njustczr** [comment]
I have the same problem...

**Comment by savan77** [comment]
I wrote a quick and ugly script to run inference but I am stuck at `center` and `scale`. As @Calmost mentioned, we may need to use other detectors here. But using other detector we will not have real time performance. 

`from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import argparse
import os
import pprint
from PIL import Image
import torch
import torch.nn.parallel
import torch.backends.cudnn as cudnn
import torch.optim
import torch.utils.data
import torch.utils.data.distributed
import torchvision.transforms as transforms

import _init_paths
from config import cfg
from config import update_config
from core.loss import JointsMSELoss
from core.function import validate
from utils.utils import create_logger
import numpy as np
import dataset
import models


args = argparse.ArgumentParser()
args.add_argument("--image",  help="path to image", default="tools/imgs/1.jpg")
args.add_argument('--cfg',
                        help='experiment configure file name',
                        required=True,
                        type=str)
args.add_argument('opts',
                        help="Modify config options using the command-line",
                        default=None,
                        nargs=argparse.REMAINDER)
args.add_argument('--modelDir',
                        help='model directory',
                        type=str,
                        default='')
args.add_argument('--logDir',
                    help='log directory',
                    type=str,
                    default='')
args.add_argument('--dataDir',
                    help='data directory',
                    type=str,
                    default='')
args.add_argument('--prevModelDir',
                    help='prev Model directory',
                    type=str,
                    default='')

args = args.parse_args()
update_config(cfg, args)
cudnn.benchmark = cfg.CUDNN.BENCHMARK
torch.backends.cudnn.deterministic = cfg.CUDNN.DETERMINISTIC
torch.backends.cudnn.enabled = cfg.CUDNN.ENABLED

model = eval('models.'+cfg.MODEL.NAME+'.get_pose_net')(
    cfg, is_train=False
)

if cfg.TEST.MODEL_FILE:
    print('=> loading model from {}'.format(cfg.TEST.MODEL_FILE))
    model.load_state_dict(torch.load(cfg.TEST.MODEL_FILE), strict=False)
    model.cuda()

criterion = JointsMSELoss(
    use_target_weight=cfg.LOSS.USE_TARGET_WEIGHT
).cuda()

# Data loading code
normalize = transforms.Normalize(
    mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]
)

transform = transforms.Compose([
        transforms.Resize((256,192)),
        transforms.ToTensor(),
        normalize,
    ])

img = Image.open(args.image)
image_tensor = transform(img) 
image_tensor = image_tensor.unsqueeze(0)

model.eval()

num_samples = 1
all_preds = np.zeros(
    (num_samples, cfg.MODEL.NUM_JOINTS, 3),
    dtype=np.float32
)

with torch.no_grad():
   
    # compute output
    outputs = model(image_tensor.type(torch.cuda.FloatTensor))
    if isinstance(outputs, list):
        output = outputs[-1]
    else:
        output = outputs
    print(output.size()) #heatmap
    #now we need to call get_final_preds(config, batch_heatmaps, center, scale)
    # but we dont have center and scale.
    
    #uncomment following line if you have center and scale.
    ### preds, maxvals = get_final_preds(cfg, output, center, scale)

    ## save_batch_heatmaps(iamge_tensor, output, "test.jpg",
    ##                    normalize=True)`

**Comment by savan77** [comment]
I wrote a quick and ugly script to run inference, but stuck at center and scale. Apparently, we need to find these details on our own, I guess. 
```

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import argparse
import os
import pprint
from PIL import Image
import torch
import torch.nn.parallel
import torch.backends.cudnn as cudnn
import torch.optim
import torch.utils.data
import torch.utils.data.distributed
import torchvision.transforms as transforms

import _init_paths
from config import cfg
from config import update_config
from core.loss import JointsMSELoss
from core.function import validate
from utils.utils import create_logger
import numpy as np
import dataset
import models


args = argparse.ArgumentParser()
args.add_argument("--image",  help="path to image", default="tools/imgs/1.jpg")
args.add_argument('--cfg',
                        help='experiment configure file name',
                        required=True,
                        type=str)
args.add_argument('opts',
                        help="Modify config options using the command-line",
                        default=None,
                        nargs=argparse.REMAINDER)
args.add_argument('--modelDir',
                        help='model directory',
                        type=str,
                        default='')
args.add_argument('--logDir',
                    help='log directory',
                    type=str,
                    default='')
args.add_argument('--dataDir',
                    help='data directory',
                    type=str,
                    default='')
args.add_argument('--prevModelDir',
                    help='prev Model directory',
                    type=str,
                    default='')

args = args.parse_args()
update_config(cfg, args)
cudnn.benchmark = cfg.CUDNN.BENCHMARK
torch.backends.cudnn.deterministic = cfg.CUDNN.DETERMINISTIC
torch.backends.cudnn.enabled = cfg.CUDNN.ENABLED

model = eval('models.'+cfg.MODEL.NAME+'.get_pose_net')(
    cfg, is_train=False
)

if cfg.TEST.MODEL_FILE:
    print('=> loading model from {}'.format(cfg.TEST.MODEL_FILE))
    model.load_state_dict(torch.load(cfg.TEST.MODEL_FILE), strict=False)
    model.cuda()

criterion = JointsMSELoss(
    use_target_weight=cfg.LOSS.USE_TARGET_WEIGHT
).cuda()

normalize = transforms.Normalize(
    mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]
)

transform = transforms.Compose([
        transforms.Resize((256,192)),
        transforms.ToTensor(),
        normalize,
    ])

img = Image.open(args.image)
image_tensor = transform(img) 
image_tensor = image_tensor.unsqueeze(0)

model.eval()

num_samples = 1
all_preds = np.zeros(
    (num_samples, cfg.MODEL.NUM_JOINTS, 3),
    dtype=np.float32
)

with torch.no_grad():
   
    # compute output
    outputs = model(image_tensor.type(torch.cuda.FloatTensor))
    if isinstance(outputs, list):
        outpuuts[-1]
        outpuuts[-1]
    else:
        output = outputs
    print(output.size()) #heatmap
    #now we need to call get_final_preds(config, batch_heatmaps, center, scale)
    # but we dont have center and scale.
    
    #uncomment following line if you have center and scale.
    ### preds, maxvals = get_final_preds(cfg, output, center, scale)

    ## save_batch_heatmaps(iamge_tensor, output, "test.jpg",
    ##                    normalize=True)
```

**Comment by eng100200** [comment]
@savan77 hello did you resolve?
